extensions/standard-processors/processors/RetryFlowFile.h

```diff
@@ -0,0 +1,121 @@
+#include "core/Processor.h"
+#include "core/ProcessSession.h"
+#include "core/Core.h"
+#include "core/Resource.h"
+#include "core/logging/LoggerConfiguration.h"
+
+namespace org {
+namespace apache {
+namespace nifi {
+namespace minifi {
+namespace processors {
+
+class RetryFlowFile : public core::Processor {
+ public:
+  explicit RetryFlowFile(std::string name, utils::Identifier uuid = utils::Identifier())
+      : Processor(name, uuid),
+        logger_(logging::LoggerFactory<RetryFlowFile>::getLogger()) {}
+  // Destructor
+  virtual ~RetryFlowFile() = default;
+  // Processor Name
+  static constexpr char const* ProcessorName = "RetryFlowFile";
+  // Supported Properties
+  static core::Property RetryAttribute;
+  static core::Property MaximumRetries;
+  static core::Property PenalizeRetries;
+  static core::Property FailOnNonNumericalOverwrite;
+  static core::Property ReuseMode;
+  // Supported Relationships
+  static core::Relationship Retry;
+  static core::Relationship RetriesExceeded;
+  static core::Relationship Failure;
+  // ReuseMode allowable values
+  static constexpr char const* FAIL_ON_REUSE = "Fail on Reuse";
+  static constexpr char const* WARN_ON_REUSE = "Warn on Reuse";
+  static constexpr char const*   RESET_REUSE = "Reset Reuse";
+
+ public:
+  bool supportsDynamicProperties() override {
+    return true;
+  }
+  /**
+   * Function that's executed when the processor is scheduled.
+   * @param context process context.
+   * @param sessionFactory process session factory that is used when creating
+   * ProcessSession objects.
+   */
+  void onSchedule(core::ProcessContext* context, core::ProcessSessionFactory* /* sessionFactory */) override;
+  /**
+   * Execution trigger for the RetryFlowFile Processor
+   * @param context processor context
+   * @param session processor session reference.
+   */
+  void onTrigger(core::ProcessContext* context, core::ProcessSession* session) override;
+
+  // Initialize, overwrite by NiFi RetryFlowFile
+  void initialize() override;
+
+
+ private:
+  void readDynamicPropertyKeys(core::ProcessContext* context);
+  // Returns (1, true) on non-numerical or out-of-bounds retry value
+  std::pair<uint64_t, bool> getRetryPropertyValue(const std::shared_ptr<FlowFileRecord>& flow_file);
+  // Returns true on fail on reuse scenario
+  bool updateUUIDMarkerAndCheckFailOnReuse(const std::shared_ptr<FlowFileRecord>& flow_file);
+  bool setRetriesExceededAttributesOnFlowFile(core::ProcessContext* context, const std::shared_ptr<FlowFileRecord>& flow_file);
+
+  std::string retry_attribute_;
+  uint64_t maximum_retries_ = 3;  // The real default value is set by the default on the MaximumRetries property
+  bool penalize_retries_ =  true;  // The real default value is set by the default on the PenalizeRetries property
+  bool fail_on_non_numerical_overwrite_ = false;  // The real default value is set by the default on the FailOnNonNumericalOverwrite property
+  std::string reuse_mode_;
```

Comment:
could be an enum

Reply:
This is definitely not worth the trouble. There is already a list of strings of allowable values, adding an enum and keeping the two sync is just an extra layer of complexity.